src/buffer_utils.cc

```diff
@@ -131,24 +131,22 @@ Buffer* create_fifo_buffer(String name, int fd, Buffer::Flags flags, bool scroll
         constexpr size_t max_loop = 16;
         size_t loop = 0;
         char data[buffer_size];
-        BufferCoord insert_coord;
+        BufferCoord insert_coord = buffer->back_coord();
         const int fifo = watcher.fd();
         do
         {
             const ssize_t count = ::read(fifo, data, buffer_size);
             if (count <= 0)
             {
                 buffer->values().erase(fifo_watcher_id); // will delete this
-                return;
+                break;
```

Comment:
This is a use after free, the comment means that the object pointer by this is deleted after the erase call, so in order to do that, this erase call needs to be postponed to the end of the function, as accessing 'buffer` will be reading into a dead lambda object.

Reply:
Oh, I see... the captures are part of the lambda object, not on the stack... right.

Updated.